### PR TITLE
Add global shuffle for data reader in object_detection.

### PR DESCRIPTION
**Pull request by yghstill**
mobile_net ssd，测试完的精度达到73.37%

### CODE REVIEW

File: PaddleCV/object_detection/reader.py [hunk 1]
```diff
@@ -287,6 +287,7 @@ def train(settings,
           enable_ce=False):
     file_path = os.path.join(settings.data_dir, file_list)
     readers = []
+    # random.seed(1)
```
Comment (qingqing01) [inline PaddleCV/object_detection/reader.py]: Please remove the commented code

File: PaddleCV/object_detection/train.py [hunk 2]
```diff
@@ -179,19 +185,19 @@ def if_exist(var):
                            predicate=if_exist)
 
     if parallel:
-        loss.persistable = True
-        build_strategy = fluid.BuildStrategy()
-        build_strategy.enable_inplace = True
-        build_strategy.memory_optimize = True
```
Comment (qingqing01) [inline PaddleCV/object_detection/train.py]: 去掉这些显存还够吗？

File: PaddleCV/object_detection/reader.py [hunk 3]
```diff
@@ -307,11 +308,11 @@ def train(settings,
                      data_dir))
     else:
         images = [line.strip() for line in open(file_path)]
+        np.random.shuffle(images)
```
Comment (qingqing01) [inline PaddleCV/object_detection/reader.py:311]: Also need to shuffle for COCO dataset.

File: PaddleCV/object_detection/mobilenet_ssd.py [hunk 4]
```diff
@@ -3,111 +3,128 @@
...
-        stride=stride,
-        padding=padding,
-        groups=num_groups,
-        act=None,
-        use_cudnn=use_cudnn,
-        param_attr=parameter_attr,
-        bias_attr=False)
-    return fluid.layers.batch_norm(input=conv, act=act)
+class MobileNet_SSD:
+    'MobileNet_SSD'
```
Comment (qingqing01) [inline PaddleCV/object_detection/mobilenet_ssd.py]: Remove this line ?

File: PaddleCV/object_detection/mobilenet_ssd.py [hunk 5]
```diff
@@ -3,111 +3,128 @@
...
+            stride=2,
+            num_groups=int(num_groups * scale),
+            padding=1)
+        return normal_conv
 
-    return mbox_locs, mbox_confs, box, box_var
+
+def build_mobilenet_ssd(img, num_classes, img_shape):
+    if img_shape[2] != 300:
+        print("Error: image shape must be 300")
```
Comment (qingqing01) [inline PaddleCV/object_detection/mobilenet_ssd.py]: Why is it limited to 300

File: PaddleCV/object_detection/reader.py (hunk 1 above)
Comment (yghstill) [reply PaddleCV/object_detection/reader.py]: done

File: PaddleCV/object_detection/train.py (hunk 2 above)
Comment (yghstill) [reply PaddleCV/object_detection/train.py]: Resumed

File: PaddleCV/object_detection/reader.py (hunk 3 above)
Comment (yghstill) [reply PaddleCV/object_detection/reader.py:311]: done

File: PaddleCV/object_detection/mobilenet_ssd.py (hunk 4 above)
Comment (yghstill) [reply PaddleCV/object_detection/mobilenet_ssd.py]: done

File: PaddleCV/object_detection/mobilenet_ssd.py (hunk 5 above)
Comment (yghstill) [reply PaddleCV/object_detection/mobilenet_ssd.py]: Deleted

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**